Add browse when search string is empty, and on load.

juniormonkey opened the pull request:
This refactors out a few methods shared between browse and search results, and calls browse() to list all local tracks when the search string is empty.

juniormonkey committed:
Add browse when search string is empty, and on load.

Lesterpig commented:
Awesome! I'll test it tonight :+1: 
Fix #19

Lesterpig commented on mopidy_party/static/controller.js:
```diff
@@ -82,44 +83,74 @@ angular.module('partyApp', [])
 
+    if(!$scope.searchField) {
+      mopidy.library.browse({
+        'uri' : 'local:directory?type=track'
```
This does not work for my testbed :grimacing: 
I'm using `mopidy-local`, do you use the `mopidy-local-sqlite` by any chance?

It works perfectly if I remove `?type=track` from the uri.

Additionally, I'm assuming that mopidy does limit the number of returned results. Is it the case?

Thanks!

juniormonkey replied:
I do use mopidy-local-sqlite. How annoying that the path to the tracks is different between mopidy-local and mopidy-sqlite. I've changed the browse to start at 'local:directory', and added a recursive browse through the 'local:directory?type=track' directory if found, to handle this case.

Also it looks like mopidy in fact doesn't limit the number of returned results, or at least if there's a limit, it's higher than the number of songs I have. I've added a "more" button to at least paginate the lookup calls and blocking UI updates.

Lesterpig reviewed (changes requested):
I've just a small issue on the browse uri. Thank you for your contribution!

juniormonkey committed:
1. start the browse at 'local:directory' and go into 'local:directory?type=track' if necessary, to support both mopidy-local and mopidy-local-sqlite
2. only mopidy.lookup() 50 tracks at once, and add a "more" button to lookup 50 more, to avoid blocking the UI while rendering many tracks.

Lesterpig commented:
Thanks again, publishing the new version!
